runtime/tests/checker/restriction_test.go

```diff
@@ -1064,3 +1064,19 @@ func TestCheckRestrictedTypeNoType(t *testing.T) {
 		)
 	})
 }
+
+func TestCheckRestrictedTypeConformanceOrder(t *testing.T) {
```

Comment:
Might be good to have the invalid case also!

Reply:
Good point, added in eb52a589828981cf1148a8465fddd3e28215fa82